### PR TITLE
test(wpt): mark timed out tests as 'failed'

### PR DESCRIPTION
**Pull request by rossilor95**
## This relates to...
#2627

## Changes
The `WPTRunner`  class inside `runner.mjs` has been refactored to provide a more informative test statistics.

### Features

- A distinction between Files and Tests is added for reporting purposes. A File can be marked as "Passed", "Failed" or "Skipped". Test statistics remains unchanged.
- WPT logs formatting has been improved with this new information
- Files that time out are marked as "Failed"

<details>
<summary>Example of File report</summary>
<br>
report_new
</details>

<details>
<summary>New log appearance</summary>
<br>
console_new
</details>

<details>
<summary>Log example for a timed out File</summary>
<br>
timeout_new
</details>

## Status

- [x] I have read and agreed to the [Developer's Certificate of Origin][cert]
- [x] Tested
- [ ] Benchmarked (**optional**)
- [x] Documented
- [x] Review ready
- [ ] In review
- [ ] Merge ready

[cert]: https://github.com/nodejs/undici/blob/main/CONTRIBUTING.md

### CODE REVIEW

File: test/wpt/runner/runner.mjs [hunk 1]
```diff
@@ -224,14 +226,27 @@ export class WPTRunner extends EventEmitter {
             signal: AbortSignal.timeout(timeout)
           })
 
-          console.log(colors(`[${finishedFiles}/${total}] PASSED - ${test}`, 'green'))
+          if (result?.subtests.every((subtest) => subtest.status === 'PASS')) {
+            this.#stats.testsPassed += 1
+            console.log(colors(`[${finishedFiles}/${total}] PASSED - ${test}`, 'green'))
+          } else {
+            this.#stats.testsFailed += 1
+            console.log(colors(`[${finishedFiles}/${total}] FAILED - ${test}`, 'red'))
```
Comment (KhafraDev) [inline test/wpt/runner/runner.mjs:234]: Tests can have expected failures, marking the entire file as failing isn't the correct behavior here.

File: test/wpt/runner/runner.mjs [hunk 2]
```diff
@@ -307,16 +322,23 @@ export class WPTRunner extends EventEmitter {
     }
 
     this.emit('completion')
-    const { completed, failed, success, expectedFailures, skipped } = this.#stats
+
+    const { testsPassed, testsFailed, testsSkipped } = this.#stats
+    console.log(
+      `Test    results for folder [${this.#folderName}]: ` +
```
Comment (KhafraDev) [inline test/wpt/runner/runner.mjs:328]: `Test` should be replaced with `File`, as this seems to be the number of files run.

File: test/wpt/runner/runner.mjs [hunk 3]
```diff
@@ -307,16 +322,23 @@ export class WPTRunner extends EventEmitter {
...
+      `completed: ${this.#files.length}, passed: ${testsPassed}, failed: ${testsFailed}, ` +
+      `skipped: ${testsSkipped}`
+    )
+
+    const { subtestsCompleted, subtestsFailed, subtestsPassed, expectedFailures } = this.#stats
```
Comment (KhafraDev) [inline test/wpt/runner/runner.mjs:333]: These technically aren't subtests, a test file can contain multiple tests (similar to other frameworks).

**Review by KhafraDev — CHANGES_REQUESTED** [review]

File: test/wpt/runner/runner.mjs (hunk 1 above)
Comment (rossilor95) [reply test/wpt/runner/runner.mjs:234]: Hi @KhafraDev and thank you for your review! Sorry, I hadn't considered this scenario. I'll try and fix that 😁

**Review by KhafraDev — APPROVED** [review]

**Review by metcoder95 — APPROVED** [review]

**Comment by rossilor95** [comment]
Hi @KhafraDev and @metcoder95, I was wondering if there is anything else I need to do for this PR. 

As far as I understood, the single CI test that failed is one of the flaky ones (it was green before I fixed the linting). If CI is the problem, can I have a second run of the failed one, please?

Thank you in advance 😄

**Pull request merged**